apis/store/v1alpha1/secretstore_types.go

```diff
@@ -0,0 +1,71 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SecretStoreSpec defines the desired state of SecretStore
+type SecretStoreSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Type=string
+	Controller string               `json:"controller,omitempty"`
+	Store      runtime.RawExtension `json:"store,omitempty"`
```

Comment:
Same here, From comment https://github.com/ContainerSolutions/externalsecret-operator/pull/119/files#r512541423

Reply:
To make the Store be another Struct that has Auth, and Parameters, and then make only the Parameters be a runtime.RawExtension, would that cause a lot of changes? We can also plan this change for later, but we have to keep an eye on it :thinking:

Reply:
It wouldn't really offer match in the store since `auth` would need to be an map `interface{}` as well.

Reply:
We would end up with something like:

```
type Store struct {
	// +kubebuilder:validation:Required
	Type       string               `json:"type"`
	Auth       runtime.RawExtension `json:"auth"`
	Parameters runtime.RawExtension `json:"parameters"`
}
```

Which offers no noticeable advantage

Reply:
Doesn't it offer the advantage to at least force this 3 fields to be required fields?